test/integration/nsmd_test_utils/test_utils.go

```diff
@@ -134,10 +138,174 @@ func DeployNSC(k8s *kube_testing.K8s, node *v1.Node, name string, timeout time.D
 	Expect(nsc.Name).To(Equal(name))
 
 	k8s.WaitLogsContains(nsc, "nsc", "nsm client: initialization is completed successfully", timeout)
+
 	logrus.Printf("NSC started done: %v", time.Since(startTime))
 	return nsc
 }
 
+func DeployAdmissionWebhook(k8s *kube_testing.K8s, name string, image string) (*appsv1.Deployment, *v1.Service) {
+	awDeployment := CreateAdmissionWebhookDeployment(k8s, name, image, "default")
+	awService := CreateAdmissionWebhookService(k8s, name, "default")
+
+	// MutatingWebhookConfiguration is still created in `prepare-circle-integration-test.sh`.
+	// When created from code a bug is reproduced: the object is properly created with the correct key for CA Bundle,
+	// but after the test execution the key is exchanged with another value causing an error in the K8s API Server:
+	// ```
+	// failed calling webhook "admission-webhook.networkservicemesh.io":
+	// Post https://nsm-admission-webhook-svc.default.svc:443/mutate?timeout=30s: x509: certificate signed by unknown authority
+	// ````
+	// This is results in not adding an init container to the required pods and test failure for empty logs from the missing container.
+	// Once this is fixed, next line can be un-commented, as well as the cleanup for MutatingWebhookConfiguration in DeleteAdmissionWebhook()
+	// Applying `admission-webhook-cfg.yaml` should be removed from `prepare-circle-integration-test.sh` too.
+
+	//awc := CreateMutatingWebhookConfiguration(k8s, name, "default")
+
+	return awDeployment, awService
+}
+
+func CreateAdmissionWebhookDeployment(k8s *kube_testing.K8s, name string, image string, namespace string) *appsv1.Deployment {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"app": "nsm-admission-webhook",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Deployment",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "nsm-admission-webhook"},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "nsm-admission-webhook",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:            name,
+							Image:           image,
+							ImagePullPolicy: v1.PullIfNotPresent,
+							VolumeMounts: []v1.VolumeMount{
+								{
+									Name:      "webhook-certs",
+									MountPath: "/etc/webhook/certs",
+									ReadOnly:  true,
+								},
+							},
+						},
+					},
+					Volumes: []v1.Volume{
+						{
+							Name: "webhook-certs",
+							VolumeSource: v1.VolumeSource{
+								Secret: &v1.SecretVolumeSource{
+									SecretName: "nsm-admission-webhook-certs",
```

Comment:
I assume that this Secret also should be created from golang. You can check how it is created in helm-deployment: https://github.com/networkservicemesh/networkservicemesh/blob/master/deployments/helm/nsm/charts/admission-webhook/templates/admission-webhook-secret.tpl